src/rust/catloop/mod.rs

```diff
@@ -285,11 +292,15 @@ impl CatloopLibOS {
         // Will use this later on close.
         let yielder_handle: YielderHandle = yielder.get_handle();
         let catmem: Rc<RefCell<CatmemLibOS>> = self.catmem.clone();
-        let next_port: u16 = self.next_port;
+        let ephemeral_ptr: Rc<RefCell<EphemeralPorts>> = self.ephemeral_ports.clone();
+        let new_port: u16 = match ephemeral_ports.alloc_any() {
+            Ok(new_port) => new_port,
+            Err(e) => return Err(e)
+        };
```

Comment:
Could you move the ephemeral port allocation logc just to just after the queue descript allocation?